### DISCUSSION TITLE
tile server with more one tile file.

### DISCUSSION THREAD

**Discussion by Maxim88S** [openmaptiles/openmaptiles#1664] [Q&A]
Good afternoon, sorry for the questions. I'm just starting to understand osm. Can you explain what settings of the quickstart file need to be changed in order to upload the second tile and so that it is displayed along with the first on the server in the styles section. And then how to ensure that both tiles are displayed on the server at the same time? Should they be combined and how can a route be built between them?  Perhaps there are some other solutions for this?

**Comment by ZeLonewolf** [comment]
I'm not quite sure what's being asked here - the question is very confusing and likely not related to openmaptiles.  I would suggest opening a thread at https://community.openstreetmap.org/ and describe what you're trying to accomplish, and you'll get responses from a wider audience.

**Comment by Maxim88S** [comment]
Good afternoon, thanks for your answer.
In general, a question about opentailserver
1.how can I make both tiles displayed simultaneously on the tileserver?
 2. Is it possible to build a route between them?
3. How can I make sure that when a vector tile is created, raster tiles are also created?

**Comment by ZeLonewolf** [comment]
Please ask your question on the OSM Community forum through the link I
posted above. It's not clear what you're trying to accomplish and I do not
think this is an openmaptiles question.

**Comment by ImreSamu** [comment]
> Can you explain what settings of the quickstart file need to be changed in order
>  to upload the second tile and so that it is displayed along with the first on the server in the styles section. 
>  And then how to ensure that both tiles are displayed on the server at the same time? Should they be combined ...

The easiest solution is to find an area that includes all the regions you need. 
This way, you avoid complex technical solutions that could affect both the data and the display.

For example, if you need Spain, Austria, Germany, and France, choose Europe as a single, contiguous area. 
Alternatively, you can extract the specific areas you need from Europe 
(  you can request help at https://community.openstreetmap.org/   or check the [osmium tools](https://osmcode.org/osmium-tool/)  ).

In short, the simplest solution to your problem is 
* to combine the necessary areas into one large  (input) region!
* alternatively, you can create a special GeoJSON file that covers the needed areas and extract those regions from the OSM planet file.

imho:  Any other solution would be more complicated and requires a deep understanding of data generation and display.

It is possible that I misunderstood your question, so please provide more information. Specifically, what is the root problem? What is your background knowledge? This way, someone might be able to give you a better answer.
Since there are various ways to display MBTiles data, it’s also worth asking on the https://community.openstreetmap.org/

**Reply by Maxim88S** [reply]
Thanks for the answer. I understood you. I will do as you said. The goal was simple - it took me a very long time to generate mbtiles for the entire region and I wanted to do it in pieces. Can you answer about raster tiles? Is it possible to make the server immediately generate raster tiles for the entire tile?

**Reply by ImreSamu** [reply]
>  How can I make sure that when a vector tile is created, raster tiles are also created?

Typically, tile servers can generate raster output from vector tiles, 
as they support both vector and raster outputs. 

You can refer to this documentation for more details: 
https://openmaptiles.org/docs/host/tileserver-gl/

![](https://openmaptiles.org/media/tileserver_gl_styles.png)

If you specifically need pre-generated PNG tiles, you'll need to use some form of post-processing. 
This can be done with a special tools - or just  a simple Python script.

However, it's important to note that vector tiles are usually generated only up to zoom level 14. 
The browser client can then use overzoom to create the necessary images up to zoom level 18, applying the appropriate style.

Generating raster tiles for the entire globe at zoom level 18 would require approximately 45 TB of storage, 
whereas vector tiles would need only about 100 GB.
